torchmetrics/functional/classification/jaccard.py

```diff
@@ -69,10 +69,10 @@ def _jaccard_from_confmat(
 def jaccard_index(
     preds: Tensor,
     target: Tensor,
+    num_classes: int,
```

Comment:
this may be breaking some use-cases, as it reorders the arguments in case a user is using positional args